agents/gce/fence_gce.py

```diff
@@ -103,6 +109,13 @@ def replace_api_uri(options, http_request):
 
 def retry_api_execute(options, http_request):
 	replaced_http_request = replace_api_uri(options, http_request)
+	action = ACTION_IDS[options["--action"]] if options["--action"] in ACTION_IDS else 0
+	try:
+		user_agent_header = USER_AGENT % (VERSION, options["--image"], action)
```

Comment:
You have to add this parameter to `define_new_opts()` and the `device_opt` array.

Reply:
Or I guess you can remove `--` from it, as it's auto-detected in your code below.

Reply:
Thanks, I removed the dashes.